ext/curl/curl_file.c

```diff
@@ -131,24 +132,13 @@ ZEND_METHOD(CURLFile, setPostFilename)
 }
 /* }}} */
 
-ZEND_BEGIN_ARG_INFO_EX(arginfo_curlfile_create, 0, 0, 1)
-	ZEND_ARG_INFO(0, filename)
-	ZEND_ARG_INFO(0, mimetype)
-	ZEND_ARG_INFO(0, postname)
-ZEND_END_ARG_INFO()
-
-ZEND_BEGIN_ARG_INFO(arginfo_curlfile_name, 0)
-	ZEND_ARG_INFO(0, name)
-ZEND_END_ARG_INFO()
-
-
 static const zend_function_entry curlfile_funcs[] = {
-	PHP_ME(CURLFile,			__construct,        arginfo_curlfile_create, ZEND_ACC_PUBLIC)
+	PHP_ME(CURLFile,			__construct,        arginfo_curl_file_create, ZEND_ACC_PUBLIC)
```

Comment:
These should not reuse arginfo from free functions. They should be declared as an explicit `class CURLFile { ... }`.

Reply:
I was not exactly sure I implemented it correctly but felt right

Reply:
It would be better to create a separate `curl_file.stub.php` file, which declares the class and then use the generated arginfo structures.

Reply:
now I start to get this whole thing 👌